client/coral-ui/components/Dropdown.js

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styles from './Dropdown.css';
+import Icon from './Icon';
+import cn from 'classnames';
+import ClickOutside from 'coral-framework/components/ClickOutside';
+
+class Dropdown extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      selected: {
+        label: props.label || '',
+        value: props.value || ''
+      },
+      isOpen: false
+    };
+  }
+  
+  componentWillReceiveProps(newProps) {
+    if (newProps.value && newProps.value !== this.state.selected.value) {
+      this.setState({
+        selected: {
+          label: newProps.label,
+          value: newProps.value
+        }
+      });
+    }
+  }
+
+  compoentDidMount() {
+    document.addEventListener('click', this.handleClick, false);
+    document.addEventListener('touchend', this.handleClick, false);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('click', this.handleClick, false);
+    document.removeEventListener('touchend', this.handleClick, false);
+  }
+
+  fireChange = (newState) => {
+    if (newState.selected !== this.state.selected && this.props.onChange) {
+      this.props.onChange(newState.selected.value);
+    }
+  }
+  
+  setValue = (value, label) => {
+    const newState = {
+      selected: {
+        label: label,
+        value: value
+      },
+      isOpen: false
+    };
+    this.fireChange(newState);
+    this.setState(newState);
+  }
+
+  handleClick = () => {
+    this.setState({
+      isOpen: !this.state.isOpen
+    });
+  }
+
+  hideMenu = () => {
+    this.setState({
+      isOpen: false
+    });
+  }
+
+  renderLabel() {
+    if (this.props.label) {
+      return this.props.label;
+    } else if (this.props.value) {
+      return this.props.value;
+    } else {
+      return this.props.placeholder;
```

Comment:
Also can't we use the labels defined for the `Option`s to get the label?

Reply:
Read the description of the PR. There are the differences.

Reply:
> re: Also would it be possible to use the labels defined for the Options to get the label?

Yes, we could. I thought you wanted it to be semantic. That's why I kept going with the `children` to have something like 

```js
<Option value={'BANNED'}>{t('community.banned')}</Option>
```

Instead of:
```
<Option value={'BANNED'} label={t('community.banned')} />
```

Reply:
Why is the latter version not semantic?